spp_api/controllers/api.py

```diff
@@ -113,14 +113,19 @@ def create_one__POST(self, namespace, version, model, **kw):
         _api_endpoint_model, methods=["GET"], type="http", auth="none", csrf=False
     )
     def read_multi__GET(self, namespace, version, model, **kw):
+        kw.get("request_id", None)
+        if "request_id" in kw:
+            del kw["request_id"]
+
         path = kw.get("path")
         del kw["path"]
-        path.search_treatment_kwargs(kw)
-
+        kw = path.search_treatment_kwargs(kw)
         records = self.get_records(path.model, kw)
+        records = records.search_read(**kw)
+
         response_data = {
-            "results": records.search_read(**kw),
-            "total": records.search_count(kw.get("domain", [])),
+            "results": records,
+            "total": len(records),
```

Comment:
@nhatnm0612 I don't quite understand what you said here but **kw** is already changed inside **path.search_treatment_kwargs(kw)** and will have a keys **["limit", "offset", "domain", "fields"]** before it reach **records.search_read(**kw)** and will become argument that will read like this ** records.search_read(limit=0, offset=0, domain=[], fields=[])**

Reply:
@reichie020212 I meant, `kw` is a reference when you passing it to the method `path.search_treatment_kwargs()`. Which means its changes inside the function also kept after the function ended, it's not a scoped variable which only available inside the scope of function [method] anymore.

Reply:
@nhatnm0612 yes the changes in **kw** will be kept even after the function ended and that changes will be used **records.search_read(**kw)**, that's intentional since I need **kw** to only have **["limit", "offset", "domain", "fields"]** as keys to be used as arguments in search_read

Reply:
@reichie020212 I mean, what needs to be done, you do it inside `path.search_treatment_kwargs()` method, to change `kw`. And those line I marked, I see no changes in the way the machine runs, only changes the code formatting. Which are unnecessary changes.

Reply:
@nhatnm0612 ah, i see, i put `records.search_read(**kw)` in variable `records` so that I can use `len()` in `records`.

I used `len()` because `search_count` doesn't have limit and offset and this results to have different counts compare to the actual records in variable `records`

For example, I have 10 records of res_partner but I added limit of 2 so in `records` variable it will have 2 records of res_partner, but in `records.search_count(kw.get("domain", []))` it will count as 10.

Reply:
I think it's better to use `search()` with `count=True`. Well, about this, I am not so sure which one is better: to add more functions to python cache or using more connection to database. But I agreed with you this.

Reply:
@nhatnm0612 let me try that.

Reply:
thanks :D resolved